modtools/item-trigger.lua

```diff
@@ -120,61 +136,85 @@ function getitemType(item)
  return itemType
 end
 
+function checkMode(table,triggerArg)
+ for _,command in ipairs(triggerArg) do
+  if command[""..table.mode..""] then
+   local reqModeType = command[""..table.mode..""]
+   local modeType = tonumber(table.modeType)
+   if #reqModeType == 1 then
+    if (tonumber(reqModeType) or tonumber(reqModeType[1])) == modeType then
+     utils.fillTable(command,table)
+     processTrigger(command)
+     utils.unfillTable(command,table)
+	end
+   elseif #reqModeType > 1 then
+    for _,r in ipairs(reqModeType) do
+     if tonumber(r) == modeType then
+      utils.fillTable(command,table)
+      processTrigger(command)
+      utils.unfillTable(command,table)
+     end
+    end
+   else
+    utils.fillTable(command,table)
+    processTrigger(command)
+    utils.unfillTable(command,table)
+   end
+  end
+ end
+end
+
 function handler(table)
  local itemMat = dfhack.matinfo.decode(table.item)
  local itemMatStr = itemMat:getToken()
  local itemType = getitemType(table.item)
  table.itemMat = itemMat
  table.itemType = itemType
 
- for _,command in ipairs(itemTriggers[itemType] or {}) do
-  if command[table.mode] then
-   utils.fillTable(command,table)
-   processTrigger(command)
-   utils.unfillTable(command,table)
-  end
- end
-
- for _,command in ipairs(materialTriggers[itemMatStr] or {}) do
-  if command[table.mode] then
-   utils.fillTable(command,table)
-   processTrigger(command)
-   utils.unfillTable(command,table)
-  end
- end
+ checkMode(table,(itemTriggers[itemType] or {}))
+ checkMode(table,(materialTriggers[itemMatStr] or {}))
 
  for _,contaminant in ipairs(table.item.contaminants or {}) do
   local contaminantMat = dfhack.matinfo.decode(contaminant.mat_type, contaminant.mat_index)
   local contaminantStr = contaminantMat:getToken()
   table.contaminantMat = contaminantMat
-  for _,command in ipairs(contaminantTriggers[contaminantStr] or {}) do
-   utils.fillTable(command,table)
-   processTrigger(command)
-   utils.unfillTable(command,table)
-  end
+  checkMode(table,(contaminantTriggers[contaminantStr] or {}))
   table.contaminantMat = nil
  end
 end
 
-function equipHandler(unit, item, isEquip)
- local mode = (isEquip and 'onEquip') or (not isEquip and 'onUnequip')
-
+function equipHandler(unit, item, mode, modeType)
  local table = {}
  table.mode = mode
+ table.modeType = modeType
  table.item = df.item.find(item)
  table.unit = df.unit.find(unit)
- if table.item and table.unit then -- they must both be not nil or errors will occur after this point with instant reactions.
+ if table.item and table.unit then--they must both be not nil or errors will occur after this point with instant reactions.
```

Comment:
Spaces around comments make them more readable. No need to remove them.

Reply:
My bad, not sure why I did that.